src/utils/react/flattenChildren.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * copied from
+ * https://github.com/grrowl/react-keyed-flatten-children/
+ * as it hasn't been updated in a while and doesnt support react v17
```

Comment:
What features of react 17/18 aren't supported?

Reply:
I faced some ESM build issues during the migration, and as I noticed the lib. is not updated from the last 2 years, I imported and did the fix.